cmd/shadowsocks-local/local.go

```diff
@@ -1,430 +1,814 @@
+			return false
+		}
+		return port != ""
+	}
+
+	for addr, pwd := range config.ServerPassword {
+		if !hasPort(addr) || pwd == "" {
+			ss.Logger.Fatal("Failed generating ciphers, server address and password mismatching", zap.String("address", addr))
+		}
+	}
 
+	return nil
+}
+
+// main locical about the local server
+func main() {
+	var configFile, ServerAddr, LocalAddr, Password, Method, MultiServerMode string
+	var ServerPort, Timeout, LocalPort, matrixport int
+	var printVer bool
+	var config *ss.Config
+	var err error
+
+	flag.BoolVar(&printVer, "v", false, "print version")
+	flag.StringVar(&configFile, "config", "", "specify config file")
+	flag.StringVar(&LocalAddr, "addr", "127.0.0.1", "local socks5 proxy serve address")
+	flag.IntVar(&LocalPort, "port", 1085, "local socks5 proxy port")
+	flag.StringVar(&ServerAddr, "saddr", "", "server address")
+	flag.IntVar(&ServerPort, "sport", 0, "server port")
+	//flag.BoolVar(&UDP, "u", false, "use the udp to serve")
+	flag.IntVar(&matrixport, "pprof", 0, "set the metrix port to Enable the pprof and matrix(TODO), keep it 0 will disable this feature")
+	flag.StringVar(&MultiServerMode, "multiserver", "fastest", "3 modes for shadowsocks local detect ss server: \n\t\tfastest: get fastest server to request\n\t\tround-robin: get server with round-robin for request\n\t\tdissable: only request for first server")
+	flag.StringVar(&Password, "passwd", "", "server password")
+	flag.IntVar(&Timeout, "timeout", 300, "timeout in seconds")
+	flag.StringVar(&Method, "method", "aes-256-cfb", "encryption method, default: aes-256-cfb")
+	flag.StringVar(&ss.Level, "level", "info", "given the logger level for ss to logout info, can be set in debug info warn error panic")
+
+	// show the help info when parse flags failed
+	flag.Parse()
+	if !flag.Parsed() {
+		flag.Usage()
+		os.Exit(0)
+	}
 	if printVer {
 		ss.PrintVersion()
 		os.Exit(0)
 	}
 
-	cmdConfig.Server = cmdServer
-	ss.SetDebug(debug)
+	// init the logger
+	ss.SetLogger()
+	ss.Logger.Info("Starting shadowsocks local server")
 
-	if strings.HasSuffix(cmdConfig.Method, "-auth") {
-		cmdConfig.Method = cmdConfig.Method[:len(cmdConfig.Method)-5]
-		cmdConfig.Auth = true
+	// set the pprof
+	if matrixport > 0 {
+		go http.ListenAndServe(":"+strconv.Itoa(matrixport), nil)
 	}
 
-	exists, err := ss.IsFileExists(configFile)
-	// If no config file in current directory, try search it in the binary directory
-	// Note there's no portable way to detect the binary directory.
-	binDir := path.Dir(os.Args[0])
-	if (!exists || err != nil) && binDir != "" && binDir != "." {
-		oldConfig := configFile
-		configFile = path.Join(binDir, "config.json")
-		log.Printf("%s not found, try config file %s\n", oldConfig, configFile)
+	// set the options for the config new
+	var opts []ss.ConfOption
+
+	// choose the encrypt method then check
+	if Method == "" {
+		// TODO change into gcm AEAD
+		Method = "aes-256-cfb"
+		opts = append(opts, ss.WithEncryptMethod("aes-256-cfb"))
```

Comment:
unnecessary line

Reply:
perhaps we'd better warning the user to set the method

Reply:
@dszhengyu fixed in 8796767